validator/client/src/main/java/tech/pegasys/artemis/validator/client/Signer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.validator.client;
+
+import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.compute_signing_root;
+import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.get_domain;
+
+import com.google.common.primitives.UnsignedLong;
+import org.apache.tuweni.bytes.Bytes;
+import tech.pegasys.artemis.datastructures.state.Fork;
+import tech.pegasys.artemis.datastructures.validator.MessageSignerService;
+import tech.pegasys.artemis.util.async.SafeFuture;
+import tech.pegasys.artemis.util.bls.BLSSignature;
+import tech.pegasys.artemis.util.config.Constants;
+
+public class Signer {
```

Comment:
This is a "richer" signing approach.  The underlying `MessageSignerService` probably needs to know more about the type of object its signing since eth2signer will have different end points for blocks, attestations and randao reveal.  But unclear at the moment if it should just accept the actual java types or if something like this class should exist to calculate the signing root and just pass through the required data.